### PR TITLE
feat: item utilities, colors

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? 

move item parsing utils (some) from arsenal-parser to warframe-items

TODO:
- [x] Eliminate all possible dependency cycles
- [x] Upstream eslint overrides
---

### Reproduction steps

---

### Considerations
- Does this contain a new dependency? **[Yes]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Feature]**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,9 +1,64 @@
...
+    }
+
     export default class Items extends Array<Item>{
         constructor(options: ItemsOptions, ...items: Item[]);
         options: ItemsOptions;
         i18n: BundleofI18nBundle<Locale>;
+        /**
+         * Utilities around items
+         * only available from ESM import
+         */;
```
Comment (TobiTenno) [inline index.d.ts:23]: whoops, i probably need to remove this part
```suggestion
```

and move it up top

File: index.d.ts [hunk 2]
```diff
@@ -1,9 +1,64 @@
+// Project: warframe-items
 declare module 'warframe-items' {
+    module 'warframe-items/utilities' {
```
Comment (TobiTenno) [inline index.d.ts:3]:
```suggestion
    /**
     * Utilities around items
     * only available from ESM import
     */
    module 'warframe-items/utilities' {
```

File: index.mjs [hunk 3]
```diff
@@ -70,15 +70,15 @@ try {
...
+const defaultCategories = readdirSync(resolve(directory, './data/json/'))
   .filter((f) => f.includes('.json'))
   .map((f) => f.replace('.json', ''))
   .filter((f) => !ignored.includes(f));
 
 const defaultOptions = { category: defaultCategories, i18n: false, i18nOnObject: false };
 
 export default class Items extends Array {
-  constructor(options, ...existingItems) {
+  constructor(options = [''], ...existingItems) {
```
Comment (TobiTenno) [inline index.mjs:81]:
```suggestion
  constructor(options, ...existingItems) {
```

experimental add, meant to remove

File: package.json [hunk 4]
```diff
@@ -5,17 +5,23 @@
...
-    "import": "./index.mjs",
-    "require": "./index.js"
+    ".": {
+      "import": "./index.mjs",
+      "require": "./index.js",
+      "types": "./index.d.ts"
+    },
+    "./utilities": {
+      "import": "./utilities/index.mjs"
+    }
```
Comment (TobiTenno) [inline package.json:15]: reformatted w/ the same overall meanings to ensure we could do both split packages for esm/cjs & do subpackage exports

File: package.json [hunk 5]
```diff
@@ -5,17 +5,23 @@
...
   },
   "scripts": {
     "build": "node ./build/build.mjs",
     "test": "nyc mocha --exclude test/*.regression.spec.mjs",
     "regression": "mocha --include test/index.regression.spec.mjs --exclude test/index.spec.mjs",
     "lint": "eslint --ignore-path .gitignore . --ext .js,.mjs,.cjs",
     "lint:fix": "eslint --ignore-path .gitignore . --fix --ext .js,.mjs,.cjs",
-    "postinstall": "install-peerdeps @wfcd/eslint-config@^1.4.1 -S",
+    "pretest": "npx npm-install-peers",
+    "postinstall": "npx -p install-peerdeps@2.0.3 install-peerdeps @wfcd/eslint-config@^1 -S",
```
Comment (TobiTenno) [inline package.json:24]: i know it's a lot of npx, but fewer dev deps!

File: package.json [hunk 6]
```diff
@@ -70,16 +77,29 @@
...
+    "rules": {
+      "import/extensions": [
+        2,
+        "never",
+        {
+          "mjs": "always",
+          "json": "always"
+        }
+      ]
+    }
```
Comment (TobiTenno) [inline package.json:93]: should be able to remove this w/ updated eslint-config
```suggestion
    }
```

File: test/lookup.mjs [hunk 7]
```diff
@@ -1,6 +1,5 @@
 /* eslint-disable no-console */
-
-import Items from '../index.mjs';
+import Items from 'warframe-items';
```
Comment (TobiTenno) [inline test/lookup.mjs:2]: yaaay internal import references to avoid weird relative paths!

**Review by SlayerOrnstein — APPROVED** [review]

**Pull request merged**